### ISSUE TITLE
Normalize data before tree building, invert before microdata

### ISSUE THREAD

**Issue by yoid2000** [diffix/syndiffix#159]
Well, to be safe, I should only normalize float, integer, and timestamp type. Bools are anyway 0 and 1, and strings turn into packed integers counting from 0, so these are well-behaved.

For each integer, float, and datetime convertor, I want to make a scaler:     scaler = `MinMaxScaler()`

In `apply_convertors()` in microdata.py, I want to do apply the scaler for each columns:   `scaler.fit_transform()`. This comes after the `_apply_convertor()` loop, which creates columns that are all floats.

Note that `_apply_convertor()` calls `convertor.to_float()`. 

In `from_interval()` is where we need to go from normalized to original. These are straightforward.

Note: Make sure that NaN values, with a doubling of the range space, is handled correctly

**Comment by yoid2000** [comment]
Hmmmm.  I think this all might be cleaner if, in microdata generation, we first convert all (normalized) intervals to (normalized) floats. Then in a separate step, we invert the normalized floats to original floats. Then after that, we do the string, time, interval conversions from the original floats.

**Comment by yoid2000** [comment]
Ok, NaN is tricky. When we normalize the data, we can retain the NaNs.

Then, in Forest, the NaN values are converted to a null_mapping value (in this case, basically 2.0). 

Then the snapped interval is 0-2 instead of 0-1. This is not a problem, just an observation.

So we need to make sure that the NaN's are restored before inverting the normalization.

Which seems to be straight-forward, because `from_interval()`is not called if the value matches the null_mapping

**Comment by yoid2000** [comment]
Implemented with #160
